source/dem/non_uniform_insertion.cc

```diff
@@ -62,25 +66,43 @@ NonUniformInsertion<dim>::insert(
         Utilities::MPI::all_gather(communicator, my_bounding_box);
 
       // Distbuting particles between processors
-      if (Utilities::MPI::this_mpi_process(communicator) !=
-          (Utilities::MPI::n_mpi_processes(communicator) - 1))
+      if (this_mpi_process != (n_mpi_process - 1))
         this->inserted_this_step_this_proc =
-          floor(this->inserted_this_step /
-                Utilities::MPI::n_mpi_processes(communicator));
+          floor(this->inserted_this_step / n_mpi_process);
       else
```

Comment:
You could replace this if/else structure by just have everyone equal to           f
foor(this->inserted_this_step / n_mpi_process); then overwriting the result with an if((this_mpi_process == (n_mpi_process - 1))). It would actually make it easier to read.

Reply:
done